src/net.rs

```diff
@@ -236,6 +240,9 @@ impl NetworkStream for HttpStream {
             Https(ref mut inner) => inner.get_mut().peer_name()
         }
     }
+
+    fn mark_dead(&mut self) {
```

Comment:
Ah, ok. You're saying this would be used if hyper had client-side keep-alive built-in.